torch/package/package_exporter.py

```diff
@@ -588,6 +709,112 @@ def _write(self, filename, str_or_bytes):
             str_or_bytes = str_or_bytes.encode("utf-8")
         self.zip_file.write_record(filename, str_or_bytes, len(str_or_bytes))
 
+    def _validate_dependency_graph(self):
+        # 1. No modules should be denied.
+        # 2. No broken modules (we should have been able to retrieve source for everything interned).
+        # 3. All modules should have an associated action.
+        # 4. All patterns for which allow_empty=False have been matched at least once.
+        denied = set()
+        broken = {}
+        unhandled = set()
+        for module_name, attrs in self.dependency_graph.nodes.items():
+            if attrs.get("action") == _ModuleProviderAction.DENY:
```

Comment:
Thoughts on creating constants for these node attribute names? We're using `get` here, so any new code that is added that makes a type won't produce a `KeyError` and might take longer to debug than necessary.

Reply:
yeah, I plan to actually factor attribute data into a real class and actually have some structure, but I want some practical experience on what that structure will look like first.